tests/unit/compiler/test_replace_child_and_update_node.py

```diff
@@ -455,7 +455,8 @@ def test_select_statement(
     new_replaced_plan = plan.children_plan_nodes[0]
     assert isinstance(new_replaced_plan, SelectSnowflakePlan)
     assert new_replaced_plan._snowflake_plan.source_plan == new_plan
-    assert new_replaced_plan.analyzer == mock_query_generator
+    # new_replaced_plan is created with QueryGenerator.to_selectable
+    assert new_replaced_plan.analyzer == mock_analyzer
```

Comment:
is this change to pass the new "setter" check?

Reply:
`new_replaced_plan` is created by calling within `QueryGenerator.to_selectable` which uses `self` as an analyzer. 

Before this change `Selectable.analyzer` would have returned the analyzer used to create the selectable i.e. `mock_query_generator`.

After this change, `Selectable.analyzer` will return the appropriate `session.analyzer`. Therefore, we need to update this test.